sql/partitioning.sql

```diff
@@ -1,4 +1,8 @@
--- our default partitioning function.
--- returns a hash of val modulous the mod_factor
-CREATE OR REPLACE FUNCTION _timescaledb_internal.get_partition_for_key(val text) RETURNS int
-	AS '$libdir/timescaledb', 'get_partition_for_key' LANGUAGE C IMMUTABLE STRICT;
+-- Deprecated partition hash function
+CREATE OR REPLACE FUNCTION _timescaledb_internal.get_partition_for_key(val text)
```

Comment:
Can't we just define `CREATE OR REPLACE FUNCTION _timescaledb_internal.get_partition_for_key_any(val anyelement)` which calls `_timescaledb_internal.get_partition_for_key(val text)` and change existing partition functions to `_timescaledb_internal.get_partition_for_key_any` in an update script to avoid having to support legacy APIs in code? It seems there are a lot of IF (param == text) stuff in the code.

Reply:
There are two checks in the for 'text' input. Not sure that is "a lot" by any measure. I would very much like to get rid of the old stuff, but unless we are prepared to convert the CHECK constraints on every chunk in every table, we need to support the legacy stuff. I think we should probably avoid touching data tables if we don't need to.

Reply:
Do we need to change the CHECK constraint on existing tables?

Reply:
That's what I wanted to avoid. Are we OK with potentially having a mismatch between the configured function in the metadata catalog and the CHECK constraints? How will that affect the constraint exclusion stuff and our optimization for, e.g., `WHERE device = 1`? Won't that break when the function we put in the query expression is different from the one on the CHECK constraint?